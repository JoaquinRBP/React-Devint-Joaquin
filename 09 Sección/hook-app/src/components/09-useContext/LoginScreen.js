import React, {useContext} from 'react';
import {UserContext} from './UserContext';

export const LoginScreen = () => {
    const {user,setuser} = useContext(UserContext);
    const usuarioLogin ={
        id: 8,
        username: 'Joaquin',
        password: '123456'
    }
return (
    <>
        <h1>LoginScreen</h1>
        <hr />

        <pre>
            {JSON.stringify(user,null, 3)}
        </pre>
        <button 
        className="btn btn-primary"
        onClick={ ()=> setuser(usuarioLogin)}>
            Login
        </button>
    </>
)
}
