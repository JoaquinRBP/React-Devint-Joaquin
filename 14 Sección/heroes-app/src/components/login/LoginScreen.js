import React from 'react';
import {useNavigate} from 'react-router-dom';
export const LoginScreen = () => {
    const history = useNavigate();
    const handleLogin =()=>{
        history('/',{replace: true});
    }
return (
    <div className="container mt-5">
        <h1>LoginScreen</h1>
        <hr />

        <button 
            className="btn btn-primary"
            onClick={ handleLogin }
        >
            Login
        </button>
    </div>
)
}
