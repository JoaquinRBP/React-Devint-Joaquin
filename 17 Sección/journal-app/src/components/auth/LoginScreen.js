import React from 'react'
import { Link } from 'react-router-dom'
import {useForm} from '../../hooks/useForm';
import {useDispatch} from 'react-redux';
import validator from 'validator';
import { startGoogleLogin, startLoginEmailPassword } from '../../actions/auth';
import { EvalError, removeError } from '../../actions/ui';
import { useSelector } from 'react-redux';
export const LoginScreen = () => {
    //useDispatch -- hook
    const dispatch =useDispatch();
    const [values, handleInputChange] = useForm({
        email:'joaquin@gmail.com',
        password:123456
    });
    const {loading} = useSelector( state => state.ui);
    const {email, password} = values;
    const handleLogin = (e) => {
        e.preventDefault();
        if(isFormValid()){
            dispatch(startLoginEmailPassword(email,password));
        }
    }
    const handleGoogleLogin = (e) => {
        e.preventDefault();
        dispatch(startGoogleLogin());
    }
    const isFormValid = () => {
        if(!validator.isEmail(email)){
            dispatch(EvalError('Email is not valid'));
            //console.log('');
            return false;
        }else if(password.length<5){
            dispatch(EvalError('Password is not valid'));
            //console.log('Password is not valid');
            return false;
        }
            dispatch(removeError());
        return true;
    }
return (
    <>
        <h3 className="auth__title">LoginScreen</h3>
        <form 
        className="animate__animated animate__fadeIn animate__faster"
        onSubmit={handleLogin}>
            <input 
                type="text"
                placeholder="Email"
                name="email"
                className="auth__input"
                value={email}
                onChange={handleInputChange}
            />
            <input 
                type="pasword"
                placeholder="Password"
                name="password"
                className="auth__input"
                autoComplete="off"
                value={password}
                onChange={handleInputChange}
            />
            <button 
                type="submit"
                className="btn btn-primary btn-block"
                disabled={loading}
            >
                Login
            </button>
            <div className="auth__social-networks">
                <p>Login with Social Networks</p>
                    <div 
                        className="google-btn"
                        onClick={handleGoogleLogin}
                    >
                    <div className="google-icon-wrapper">
                        <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
                    </div>
                    <p className="btn-text">
                        <b>Sign in with google</b>
                    </p>
                    </div>
            </div>
            <Link className="link" to="/auth/register">
                Create a new account
            </Link>
        </form>
    </>
)
}
