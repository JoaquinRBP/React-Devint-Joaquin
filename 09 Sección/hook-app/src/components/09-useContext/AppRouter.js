import React from 'react'
import {
    BrowserRouter as Router,
    Route,
    Routes
    } from "react-router-dom";
import {AboutScreen} from "./AboutScreen";
import {LoginScreen} from "./LoginScreen";
import {HomeScreen}  from "./HomeScreen";
import {NavBar} from "./NavBar";
export const AppRouter = () => {
    //Routes vs Switch
    //Element vs Component
    //Redirect vs NoMatch
return (
    <Router>
        <div>
            <NavBar />
            <div className="container">
            <Routes>
                <Route exact path="/" element={<HomeScreen />} />
                <Route exact path="/about" element={<AboutScreen />} />
                <Route exact path="/login" element={<LoginScreen />} />
                <Route path='*' element={<main>
                    <h1>ERR 404</h1>
                </main>} />
            </Routes>
            </div>
        </div>
    </Router>
)
}
