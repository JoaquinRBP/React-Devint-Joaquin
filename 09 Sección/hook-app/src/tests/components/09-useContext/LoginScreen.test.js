import React from 'react';
import {UserContext} from '../../../components/09-useContext/UserContext';
import {LoginScreen} from '../../../components/09-useContext/LoginScreen';
import { mount } from 'enzyme';
describe('Pruebas sobre el LoginScreen', () => {
    const setuser = jest.fn();
    const wrapper = mount(
    <UserContext.Provider value={{setuser}}>
        <LoginScreen />
    </UserContext.Provider>);
    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('debe de ejecutar el setUser con el argumento esperado', () => {
        wrapper.find('button').prop('onClick')();
        expect(setuser).toHaveBeenCalledWith({
            id:8,
            password: '123456',
            username: 'Joaquin'
        });   
    });
})
