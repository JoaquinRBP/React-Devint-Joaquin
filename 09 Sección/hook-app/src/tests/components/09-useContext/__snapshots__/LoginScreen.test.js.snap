// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el LoginScreen debe de mostrarse correctamente 1`] = `
Array [
  <h1>
    LoginScreen
  </h1>,
  <hr />,
  <pre />,
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>,
]
`;
